code/src/containers/forms/new-product-form/NewProductForm.test.tsx

```diff
@@ -0,0 +1,328 @@
+import { act, fireEvent, screen } from "@testing-library/react";
+
+import NewProductForm from "@/containers/forms/new-product-form/NewProductForm";
+import { defaultValues } from "@/containers/forms/new-product-form/NewProductForm.constants";
+
+import routes from "@/constants/routes";
+import renderWithProviders from "@/utils/render-with-providers/renderWithProviders";
+import typeIntoInput from "@/utils/type-into-input/typeIntoInput";
+
+const mockUnwrap = jest.fn();
+
+const mockCreateProductMutation = jest.fn(() => ({ unwrap: mockUnwrap }));
+const mockShowSnackbar = jest.fn();
+const mockNavigate = jest.fn();
+
+jest.mock("@/store/api/productsApi", () => ({
+  useCreateProductMutation: () => [
+    mockCreateProductMutation,
+    { isLoading: false }
+  ]
+}));
+
+jest.mock("@/hooks/use-snackbar/useSnackbar", () => ({
+  __esModule: true,
+  default: () => ({ openSnackbarWithTimeout: mockShowSnackbar })
+}));
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => mockNavigate
+}));
+
+const testData = {
+  status: "HIDDEN",
+  image: "https://example.com",
+  quantity: 10,
+  price: 100,
+  tagIds: [3],
+  productTranslations: [
+    {
+      name: "name",
+      description: "description-description",
+      languageCode: "en"
+    }
+  ]
+};
+
+const twoTranslationsTestData = {
+  ...testData,
+  productTranslations: [
+    testData.productTranslations[0],
+    { ...testData.productTranslations[0], languageCode: "uk" }
+  ]
+};
+
+describe("Test NewProductForm", () => {
+  let imgUrlInput: HTMLInputElement;
+  let priceInput: HTMLInputElement;
+  let quantityInput: HTMLInputElement;
+  let categorySelect: HTMLSpanElement;
+  let nameInput: HTMLInputElement;
+  let descriptionInput: HTMLTextAreaElement;
+  let submitButton: HTMLButtonElement;
+  let statusInput: HTMLInputElement;
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const render = () => {
+    const result = renderWithProviders(<NewProductForm />);
+
+    imgUrlInput = screen
+      .getByTestId("new-product-image-input")
+      .querySelector("input")!;
```

Comment:
I think that we can create simple function to reuse it because it look duplicated

Reply:
Thanks, I made a function for it